.github/workflows/benchmark_visualization.yml

```diff
@@ -0,0 +1,102 @@
+name: Visualize Benchmark and upload results
+
+on:
+  push:
+    branches: [ main ]
+    paths:
+      - '**'
+      - '!docs/**' # ignore docs changes
+      - '!**.md' # ignore markdown changes
+
+permissions:
+  contents: write
+  deployments: write
+      
+jobs:
+  benchmark:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Run Benchmarks on PR Merge
```

Comment:
This name seems to be on the wrong step. This step just checks out the code.

Reply:
You're right, Updated along with some better naming for the other steps.